### PR TITLE
Htable updates

### PR DESCRIPTION
**Pull request by daviditen**
Update HTable test for 0-based indices and deprecated numElements to get it
working with 1.22.

Also moved it into the HDF5/ test directory.

### CODE REVIEW

File: test/library/packages/HDF5/htable/HTable.chpl [hunk 1]
```diff
@@ -205,30 +205,30 @@ module HTable {
       this.complete();
 
       var r : R;
-      for param ii in 1..nFields {
+      for param ii in 0..nFields-1 {
```
Comment (ben-albrecht) [inline test/library/packages/HDF5/htable/HTable.chpl:208]:
```suggestion
      for param ii in 0..<nFields {
```

File: test/library/packages/HDF5/htable/HTable.chpl [hunk 2]
```diff
@@ -245,8 +245,8 @@ module HTable {
 
     /* De-initializer. Cleans up any types the routine created. */
     proc deinit() {
-      for param ii in 1..nFields {
-        if ownedtypes[ii] then H5Tclose(types[ii-1]);
+      for param ii in 0..nFields-1 {
```
Comment (ben-albrecht) [inline test/library/packages/HDF5/htable/HTable.chpl:248]:
```suggestion
      for param ii in 0..<nFields {
```

File: test/library/packages/HDF5/htable/HTable.chpl [hunk 3]
```diff
@@ -191,7 +191,7 @@ module HTable {
      `c_array` types fall into this class). If so, this
      flags that these need to be deallocated at destruction.
     */
-    var ownedtypes : [1..nFields] bool;
+    var ownedtypes : [0..#nFields] bool;
```
Comment (ben-albrecht) [inline test/library/packages/HDF5/htable/HTable.chpl:194]: Could use `<` here or `#` in the below suggestions, to stay consistent.

**Comment by daviditen** [comment]
Thanks! I updated with your suggested changes.

**Pull request merged**